### ISSUE TITLE
Add swagger test for when "cached calls" is turned on

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/openFEC#3014]
Something like: 
```
    def test_swagger_with_cache(self):

        os.environ['CACHE_ALL_REQUESTS'] = True
        try:
            utils.validate_swagger(spec)
        except exceptions.SwaggerError as error:
            self.fail(str(error))
```
Note: This is passing when we expect it to fail - need to figure out why.

**Comment by AmyKort** [comment]
ice boxing until Swagger tests are complete.

**Comment by pkfec** [comment]
**Issue before**:
To enable cache calls on local, we have to the set the **export CACHE_ALL_REQUEST=True**. By default the variable value is set to False. In the past, we were caching all api endpoints besides non api url's **(ex: http://127.0.0.1:5000/)** which led the swagger (OpenFEC API Documentation) failures.

Now that we have refactored the **Cache Calls** logic, which saves only the api endpoints, the swagger loads fine now.

**Comment by ccostino** [comment]
@pkfec, you're right that we refactored the cached calls logic to only cache the API endpoints themselves and this fixes the problem that we had with the existing test.  However, refactoring that test a bit to provide this added check of the documentation loading with cached calls being set to `True` to turn it on helps ensure that if we ever experience a regression in the code, it'll be caught.

Furthermore, we should be adding a few more tests around the cached calls themselves to test that the caching is working correctly.  What I mean by this is given a handful of URLs, some API endpoints and some not, a test or set of tests could verify that the API endpoint calls are cached and others are not, and we could test that none of the URLs are cached if the cached calls functionality is disabled.  There could be one or two other things to test as well, this is just a start.

We went ahead without adding these tests before because of how long the feature had been in development and as a team we decided not to delay things further.  The other decision that was made in response to this was that tests would be written in the future to account for the changes and new feature, and that's what this issue encompasses.

Does this make sense?  If there are any questions or things still aren't clear please ask!

**Comment by pkfec** [comment]
Swagger Test:
Setting the cache-calls env variable to True or False, has no impact on Swagger API documentation.  Swagger Test is only looking if the the variable is EXISTS or NOT, but not necessary checking the value of the variable. There is no way that the swagger API is going to break, with the way it is right now.
Thats my concern. 

Additional Tests:
I am with you on writing additional tests to check if caching happens for API endpoints and non api endpoints. This should be tracked in a different issue ,i guess. But i am happy to write these test anytime.

**Comment by pkfec** [comment]
The swagger documentation failed to load earlier because, the URL we were trying to cache did not have a V1 in it. The cache calls logic, is looking for URL's specially with **v1**.

For Example, http://localhost:5000 is the default URL to load the swagger API documentation. Since there is no V1 in the URL, the cache calls logic throws an error which lead to the swagger API documentation to fail. I think this is when we came up with this test case.

May be i am not seeing things that way you do. I do not see a need for the swagger test, with the way cache calls URL's are filtered right now. would like to have a open discussion with your on this swagger test.

**Comment by ccostino** [comment]
Yes, that is correct about the URLs and what had happened.  What I am saying is we should still have a test to make sure we are accounting for that regression as it had occurred before so we don't get caught with it again in the future. :-)

In any case, I agree, an open discussion at this point is a good idea so we can clear up confusion and figure out the best path forward!

**Comment by AmyKort** [comment]
Is the attached PR still in need of review?

**Comment by ccostino** [comment]
@AmyKort Yes, there is still work to be done, sorry if that wasn't clear; I know this is a bit on hold due to current sprint capacity, but I've also caught up with @annalee to provide some additional help around this. :-)

**Comment by vrajmohan** [comment]
Removing from 6.2 milestone while the blocking issue #3078 is completed.

**Comment by vrajmohan** [comment]
This can be closed due to #3235.
